File: OOPROG/Sortation_Algorithms/BubbleSort.java
public class BubbleSort{
    public static void main(String[] args) {
        int[] arr = {7,3,4,1,0};

        System.out.println("Unsorted Array: ");
        printArray(arr);

        sort(arr);

        System.out.println("Sorted array: ");
        printArray(arr);
    }
    static void sort(int[] num){
        int temp, size;
        boolean swapped;

        size = num.length;

        for (int i = 0; i < size; i++) {
            swapped = false;

            for (int index = 0; index < size - i - 1; index++) {
                if (num[index] > num[index + 1]) {
                    temp = num[index + 1];
                    num[index + 1] = num[index];
                    num[index] = temp;
                    swapped = true;
                }
            }

            if (swapped == false) break;
        }
    }
    static void printArray(int[] num){
        for (int i : num) {
            System.out.print(i + " ");
        }
        System.out.println();
    }
}

File: OOPROG/Sortation_Algorithms/bubbleSort.c
#include <stdio.h>
#include <stdbool.h>

void swap(int *x, int *y);
void bubbleSort(int arr[], int iter);
void printArr(int arr[], int size);

int main(int argc, char const **argv){
    int size;
    int array[] = {1, 8, 2, 6, 4, 0, 9};

    size = sizeof(array) / sizeof(array[0]);

    printf("Unsorted array:\n");
    printArr(array, size);

    bubbleSort(array, size);

    printf("Sorted array:\n");
    printArr(array, size);

    return 0;
}

void swap(int *x, int *y){
    int temp = *y;

    *y = *x;
    *x = temp;
}

void bubbleSort(int arr[], int iter){
    bool swapped;

    for (int i = 0; i < iter; ++i) {
        swapped = false;

        for (int j = 0; j < iter - i - 1; ++j) {
            if (arr[j] > arr[j + 1]) {
                swap(&arr[j], &arr[j + 1]);
                swapped = true;
            }
        }

        if (swapped == false) break;
    }
}

void printArr(int arr[], int size){
    for (int i = 0; i < size; ++i) printf("%d ", arr[i]);
    printf("\n");
}